client/src/cmdlfem4x50.c

```diff
@@ -0,0 +1,720 @@
+            
+            // collect parities
+            p = 0;
+
+            for (int k = 0; k < 8; k++) {
+               
+                // row parity
+                p ^= (words[i].byte[j] >> k) & 1;
+                
+                // column parity
+                c[k] ^= (words[i].byte[j] >> (7-k)) & 1;
+            }
+            
+            // check row parities
+            words[i].rparity[j] = (words[i].row_parity[j] == p) ? true : false;
+
+            if (!words[i].rparity[j])
+                words[i].parity = false;
+        }
+        
+        // check column parities
+        words[i].col_parity = byte[i*7+5] ;
+        
+        for (int j = 0; j < 8; j++) {
+            words[i].cparity[j] = (((words[i].col_parity >> (7-j)) & 1) == c[j]) ? true : false;
+
+            if (!words[i].cparity[j])
+                words[i].parity = false;
+        }
+
+        // check stop bit
+        words[i].stopbit = byte[i*7+6] & 1;
+        
+        if (words[i].stopbit == 1)
+            words[i].stopparity = false;
+        
+    }
+}
+
+static void print_bit_table(const em4x50_word_t word) {
+    
+    // generate output in table form for each word including parities, stop
+    // bit, result of parity checks and hex notation of each row in msb/lsb
+    // notation
+    // individual parity errors will be highlighted in red
+        
+    int bit = 0;
+    char string[400] = {0};
+    char pstring[100] = {0};
+
+    // print binary data
+    for (int j = 0; j < 4; j++) {
+        
+        strcat(string, "  ");
+
+        // lsb notation
+        for (int k = 0; k < 8; k++) {
+            sprintf(pstring, "%i", (word.byte[j] >> (7-k)) & 1);
+            strcat(string, pstring);
+        }
+
+        strcat(string, " | ");
+
+        // binary row parities + hex bytes of word
+        sprintf(pstring, (word.rparity[j]) ? "%i" : _RED_("%i"), word.row_parity[j]);
+        strcat(string, pstring);
+        
+        if (j == 0)
+            sprintf(pstring, "  msb: 0x%02x  lsb: 0x%02x", word.byte[j], msb2lsb(word.byte[j]));
+        else
+            sprintf(pstring, "       0x%02x       0x%02x", word.byte[j], msb2lsb(word.byte[j]));
+        
+        strcat(string, pstring);
+        PrintAndLogEx(NORMAL,string);
+
+        string[0] = '\0';
+    }
+
+    strcat(string, "  ------------  --------------------\n  ");
+
+    // binary column parities
+    for (int k = 0; k < 8; k++) {
+        
+        bit = (word.col_parity >> (7-k)) & 1;
+        
+        // if column parity is false -> highlight bit in red
+        sprintf(pstring, (word.cparity[k]) ? "%i" : _RED_("%i"), bit);
+        strcat(string, pstring);
+    }
+
+    // binary stop bit
+    strcat(string, " | ");
+    sprintf(pstring, (word.stopparity) ? "%i" : _RED_("%i"), word.stopbit);
+    strcat(pstring, "  parities ");
+    strcat(string, pstring);
+    
+    // parities passed/failed
+    sprintf(pstring, (word.parity) ? _GREEN_("passed") : _RED_("failed"));
```

Comment:
replace "passed" w "ok"     We are trying to unify these kinds of output in the client.

Reply:
done